### PR TITLE
handle error when getting hostname in info api

### PR DESCRIPTION
**Pull request by allencloud**
handle error when getting hostname in info api

### CODE REVIEW

**Comment by coolljt0725** [comment]
LGTM

File: daemon/info.go [hunk 1]
```diff
@@ -133,9 +133,13 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 		v.CPUSet = sysInfo.Cpuset
 	}
 
-	if hostname, err := os.Hostname(); err == nil {
-		v.Name = hostname
+	hostname := "<unknown>"
```
Comment (thaJeztah) [inline daemon/info.go]: I think this is also used by the API, and `<hostname>` is an invalid value for a "hostname". Perhaps I'm thinking to far into this, but what if a user of the API uses this value? If would be a breaking change. Perhaps we should set the "<unknown>" in the client (when *printing* the output)

Comment (allencloud) [inline daemon/info.go]: @thaJeztah 
I tried on my ubuntu server that when hostname is `<unknown>`, executing command `hostname`,the system will show `(none)`.
Seems `<unknown>` is an invalid value.

Comment (LK4D4) [inline daemon/info.go]: I'm not sure about `<unknown>` too. Maybe we should go with safer `""` value and just add log.

**Comment by allencloud** [comment]
@LK4D4 @thaJeztah 
What you said is reasonable. Safer is better.
Update the PR.

**Comment by thaJeztah** [comment]
LGTM

**Comment by thaJeztah** [comment]
@coolljt0725 PTAL

**Comment by coolljt0725** [comment]
re-LGTM

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**